docs/src/dev-guide/optimization-guide.md

```diff
@@ -0,0 +1,25 @@
+# Optimization guide
+
+This document outlines strategies and tools used to optimize the size and load time.
+
+## Bundle analysis
+
+We use [`vite-bundle-visualizer`][vite-bundle-visualizer] to analyze the output bundle. To generate
+a bundle analysis report, run:
+
+```shell
+npm run analyze:size
+```
+
+This will generate an interactive visual breakdown of bundle contents, helping identify large
+dependencies and optimization opportunities.
```

Comment:
```suggestion
This will use [`vite-bundle-visualizer`][vite-bundle-visualizer] to generate an interactive visual
breakdown of bundle contents, helping identify large dependencies and optimization opportunities.
```